connect-wallet/pages/ShowNFTs.js

```diff
@@ -0,0 +1,42 @@
+import styles from '../styles/Home.module.css'
+import { useMetaplex } from "./useMetaplex";
+import { useState } from "react";
+
+export const ShowNFTs = () => {
+    const { metaplex } = useMetaplex();
+
+    const [nft, setNft] = useState(null);
+    const onClick = async () => {
+        let myNfts = await metaplex.nfts().findAllByOwner(metaplex.identity().publicKey);
+        let randIdx = Math.floor(Math.random() * myNfts.length);
+        await myNfts[randIdx].metadataTask.run();
+        setNft(myNfts[randIdx]);
+    };
+
+    return (
+      <div>
+        <div>
+          <div className={styles.container}>
```

Comment:
I think it might be a could idea to hide a big portion of that UI when the wallet is not connected. Instead let's display a quick message that says: "Connect your wallet to get started...".

You can use again `const { connected } = useWallet()` here for that purpose.